internal/middleware/log.go

```diff
@@ -8,29 +8,19 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+func isType[T any](typ any) bool {
+	_, ok := typ.(*T)
+	return ok
+}
+
 // LifecycleLogger is a middleware which logs lifecycle messages (started, stopped...).
 func LifecycleLogger() actor.ReceiverMiddleware {
 	return func(next actor.ReceiverFunc) actor.ReceiverFunc {
 		return func(context actor.ReceiverContext, env *actor.MessageEnvelope) {
-			accepted := func() bool {
-				if _, ok := env.Message.(*actor.Started); ok {
-					return true
-				}
-
-				if _, ok := env.Message.(*actor.Stopping); ok {
-					return true
-				}
-
-				if _, ok := env.Message.(*actor.Stopped); ok {
-					return true
-				}
-
-				if _, ok := env.Message.(*actor.Restarting); ok {
-					return true
-				}
-
-				return false
-			}()
+			accepted := isType[actor.Started](env.Message) ||
+				isType[actor.Stopping](env.Message) ||
+				isType[actor.Stopped](env.Message) ||
+				isType[actor.Restarting](env.Message)
```

Comment:
What i did is following: the `state` is carried by `t`, such as in `state = actor.Started`. From this, I’m extracting only the last part, converted to lowercase, to create a message string, like `actor <init/sys/logger> started`.

Using an _enum_ could simplify this, but it would require defining a variant for each supported message (as you suggested), along with one method for retrieving the string representation of the type and another for the label. That’s a fair amount of work, imho. However, I'm agree that it would improve performance by avoiding string manipulation...